pymongo/srv_resolver.py

```diff
@@ -15,19 +15,18 @@
 """Support for resolving hosts and options from mongodb+srv:// URIs."""
 from __future__ import annotations
 
+import importlib.util
 import ipaddress
 import random
-from typing import Any, Optional, Union
+from typing import TYPE_CHECKING, Any, Optional, Union
 
 from pymongo.common import CONNECT_TIMEOUT
 from pymongo.errors import ConfigurationError
 
-try:
+if TYPE_CHECKING:
     from dns import resolver
 
-    _HAVE_DNSPYTHON = True
-except ImportError:
-    _HAVE_DNSPYTHON = False
+_HAVE_DNSPYTHON = importlib.util.find_spec("dns") is not None
```

Comment:
Similar comment about making this a function and avoiding importlib.

Reply:
Done